packages/token/src/erc6909/erc6909.cairo

```diff
@@ -0,0 +1,347 @@
+            let mut src5_component = get_dep_component_mut!(ref self, SRC5);
+            src5_component.register_interface(interface::IERC6909_ID);
+        }
+
+        /// Creates a `value` amount of tokens and assigns them to `account`.
+        ///
+        /// Requirements:
+        ///
+        /// - `receiver` is not the zero address.
+        ///
+        /// Emits a `Transfer` event with `from` set to the zero address.
+        fn _mint(
+            ref self: ComponentState<TContractState>,
+            receiver: ContractAddress,
+            id: u256,
+            amount: u256,
+        ) {
+            assert(receiver.is_non_zero(), Errors::INVALID_RECEIVER);
+            self._update(Zero::zero(), receiver, id, amount);
+        }
+
+        /// Destroys `amount` of tokens from `account`.
+        ///
+        /// Requirements:
+        ///
+        /// - `account` is not the zero address.
+        /// - `account` must have at least a balance of `amount`.
+        ///
+        /// Emits a `Transfer` event with `to` set to the zero address.
+        fn _burn(
+            ref self: ComponentState<TContractState>,
+            account: ContractAddress,
+            id: u256,
+            amount: u256,
+        ) {
+            assert(account.is_non_zero(), Errors::INVALID_SENDER);
+            self._update(account, Zero::zero(), id, amount);
+        }
+
+        /// Transfers an `amount` of tokens from `sender` to `receiver`, or alternatively mints (or
+        /// burns) if `sender` (or `receiver`) is the zero address.
+        ///
+        /// This function can be extended using the `before_update` and `after_update` hooks.
+        /// The implementation does not keep track of individual token supplies and this logic is
+        /// left to the extensions instead.
+        ///
+        /// Emits a `Transfer` event.
+        fn _update(
+            ref self: ComponentState<TContractState>,
+            sender: ContractAddress,
+            receiver: ContractAddress,
+            id: u256,
+            amount: u256,
+        ) {
+            Hooks::before_update(ref self, sender, receiver, id, amount);
+
+            if (sender.is_non_zero()) {
+                let sender_balance = self.ERC6909_balances.read((sender, id));
+                assert(sender_balance >= amount, Errors::INSUFFICIENT_BALANCE);
+                self.ERC6909_balances.write((sender, id), sender_balance - amount);
+            }
+
+            if (receiver.is_non_zero()) {
+                let receiver_balance = self.ERC6909_balances.read((receiver, id));
+                self.ERC6909_balances.write((receiver, id), receiver_balance + amount);
+            }
+
+            self.emit(Transfer { caller: get_caller_address(), sender, receiver, id, amount });
+
+            Hooks::after_update(ref self, sender, receiver, id, amount);
+        }
+
+        /// Sets or unsets a spender as an operator for the caller.
+        fn _set_operator(
```

Comment:
Based on [Solidity reference](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC6909/ERC6909.sol#L194-L203), `_set_operator` should perform a check on `owner` and `spender` being non zero address.